### ISSUE TITLE
Performance on Cityscapes

### ISSUE THREAD

**Issue by edizhuang** [yassouali/pytorch_segmentation#16]
Hi Yassine,

I trained Deeplab V3+ (80 epochs) and got Mean_IoU 0.6310 (Pixel_Accuracy : 0.828).
I trained PSPNet (75 epochs so far) and got Mean_IoU 0.6330 (Pixel_Accuracy : 0.827).         
All with with ResNet101 and "base_size": 1024, "crop_size": 512. It seems lower than the performance they reported in the paper (PSPNet, DeepLabV3+). 

If I remove "base_size": 1024, will it use the original input size for training? I got Deeplab V3+ in this case with Mean_IoU 0.6470 (Pixel_Accuracy : 0.83). 

Do you think this performance is right? And how can I continue training with coarse mode? Thanks!

**Comment by edizhuang** [comment]
My config file is almost the same as you suggested for input size, and the rest parameters are following the example you give:
{
    "name": "DeepLab",
    "n_gpu": 2,
    
    "arch": {
        "type": "DeepLab",
        "args": {
            "backbone": "resnet101",
            "freeze_bn": false,
            "freeze_backbone": false
        }
    },

    "train_loader": {
        "type": "CityScapes",
        "args":{
            "data_dir": "data/Cityscapes",
            "batch_size": 8,
            "base_size": 1024,
            "crop_size": 512,
            "augment": true,
            "shuffle": true,
            "scale": true,
            "flip": true,
            "rotate": true,
            "blur": false,
            "split": "train",
            "mode": "fine",
            "num_workers": 8
        }
    },

    "val_loader": {
        "type": "CityScapes",
        "args":{
            "data_dir": "data/Cityscapes",
            "batch_size": 1,
            "val": true,
            "split": "val",
            "mode": "fine",
            "num_workers": 4
        }
    },

    "optimizer": {
        "type": "SGD",
        "differential_lr": true,
        "args":{
            "lr": 0.01,
            "weight_decay": 1e-4,
            "momentum": 0.9
        }
    },

    "loss": "CrossEntropyLoss2d",
    "ignore_index": 255,
    "lr_scheduler": {
        "type": "Poly",
        "args": {}
    },

    "trainer": {
        "epochs": 80,
        "save_dir": "saved/",
        "save_period": 10,
  
        "monitor": "max Mean_IoU",
        "early_stop": 10,
        
        "tensorboard": true,
        "log_dir": "saved/runs",
        "log_per_iter": 20,

        "val": true,
        "val_per_epochs": 5
    }
}

**Comment by yassouali** [comment]
Hi;

For CityScapes, you're right, the IoU should be in the range of 75-80.

CItyscapes is finely annotated (the majority of the image pixel are annotated), so you'll need to train for a lot longer than in say Pascal VOC, for example in the PSPNet paper, they trained it for 90K iterations (with a btach of 16 and 3K training images, that's ~ 480 epochs).

So if you want to get close to the reported results:
- Train for say 400 epochs using base size of 1024.
- Then finetune using a bigger base & crop size (say 2048 - 1024) for another 100 epochs.
- Then do multi-scale evaluation (`inference.py`)

If you want even more performances, use the coarse lables (`mode:coarse`), but this will only give +3 points and will take a lot more time to train. If really want to get the exact results, you'll also need to use synchornized batch norm to get ~4/3 points, use [this](https://github.com/vacancy/Synchronized-BatchNorm-PyTorch) PyTorch implementation, you'll need to cal `convert_model`, and `DataParallelWithCallback`. Maybe i'll add this option in the coming days.

**Comment by yassouali** [comment]
Just added `synch_bn` and the option `use_synch_bn` in the config file, set it to true if you want to use if, note that it'll take some additional GRAM.
